### PR TITLE
Add `DefaultErrorDescription` environment key;  Localization features; restructure project; fix spelling

### PR DESCRIPTION
**Pull request by Supereg**
# Add `DefaultErrorDescription` environment key; Localization features; restructure project; fix spelling 

## :recycle: Current situation & Problem

The current situation is as follows:
* The project folders are loosely structured.
* Some spelling issues in docs (and also public code :/).

Further, the current API design imposes some inconveniences or could be improved in regard to usability and flexibility.
* Working with `AnyLocalizedError` always requires the view using it to request a default error description. This is tedious e.g., when creating smaller, reusable view components as it clutters the initializer with configuration values. Also a view that reuses several components that rely on the same default error description has the overhead of passing the value to every single subview.
* `FieldLocalization` localizes its content instantly.

## :bulb: Proposed solution
This PR addresses the above-introduced situation as follows:
* It introduces a broadly categorized folder structure with `Views`, `ViewModifier`, `Environment` (e.g., EnvironmentKeys) and more openly defined categories `Model` (any Models or ViewModel related types) or `Utilities` (extensions or otherwise helpful types).
* Fixes the spelling mistakes
* Removed unused code in the UITests
* Creates the `DefaultErrorDescription` EnvironmentKey to resolve the above introduced shortcomings. Additionally, we introduce a global default error description.
* Create a `FieldLocalizationResource` that is based on `LocalizationStringResource` instead of localized strings for greater flexibility.

## :gear: Release Notes 
* Added the `.defaultErrorDescription` `EnvironmentKey` that can be used for view components creating `AnyLocalizedError` to source the `defaultErrorDescription` from the environment. Equally, views can set the `.defaultErrorDescription` `EnvironmentKey` to pass down customized configuration options.
* Introduce `FieldLocalizationResource`
* `StringProtocol.localized` was deprecated and replaced with `StringProtocol.localized()` to better communicate its effects

**The PR introduces the following breaking changes:**
* The `Label` initializer exposed a spelling mistake in its `textAlignment` (previously `textAllignment`) external property name. Update your code to the corrected name.
* `FieldLocalization` was replaced by `FieldLocalizationResource`. Typical SwiftUI initializers should be able to handle the update types (e.g. replacing String with LocalizationStringResource), otherwise we ship a convenience `localizedString()` extension for `LocalizationStringResource` to easily convert to a String.

### Related PRs
--

### Testing
New UI tests were added. The TestApp was updated to adapt to the breaking changes.

### Reviewer Nudging
Look altho e`DefaultErrorDescription` type and at `AnyLocalizedError`. Refer to the `DefaultErrorDescriptiontestView` to see how it is used.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Supereg** [comment]
Ping @PSchmiedmayer CI help 🙈

**Comment by Supereg** [comment]
I'm at a point where I think I have fixed some issue with a timeout on the HTTP view tests, but am now at a point where the CI is completely broken. Can't event view the `TestApp.xresult` anymore because it seems to be incomplete 🙃

File: Sources/SpeziViews/Model/FieldLocalizationResource.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+/// A ``FieldLocalizationResource`` describes a localization of a `TextField` instance using a ``FieldLocalizationResource/title`` and ``FieldLocalizationResource/placeholder``.
+public struct FieldLocalizationResource: Codable {
```
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Model/FieldLocalizationResource.swift:13]: Nice, that looks good 👍

File: Sources/SpeziViews/Environment/DefaultErrorDescription.swift
Comment (PSchmiedmayer) [inline Sources/SpeziViews/Environment/DefaultErrorDescription.swift:1]: Nice, l like this idea!

**Review by PSchmiedmayer — APPROVED** [review]
Thank you for the improvements, this looks great!

**Comment by PSchmiedmayer** [comment]
@Supereg Yes, seems like the simulator is not starting which is strange. I will take a look at that ...

**Comment by PSchmiedmayer** [comment]
@Supereg restated the build agents and restarted everything. Seems to pass now 👍

**Pull request merged**